### PR TITLE
fix: Button with icon styling

### PR DESCRIPTION
**Pull request by johnkim-det**
## Description

Styling for buttons was dependent on usage of Antd icons. Adding classes to support using both Antd icons and our `Icon` component in the `Button`'s `icon` prop.

## Test Plan

No release testing required. Regression testing on Buttons. Design Kit page should show examples with Antd icons and font icons

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

WEB-1226

### CODE REVIEW

File: webui/react/src/pages/DesignKit.tsx [hunk 1]
```diff
@@ -246,8 +246,16 @@ const ButtonsSection: React.FC = () => {
         </Space>
         <hr />
         <strong>Default Button with icon</strong>
+        With Antd Icon
```
Comment (ClaireNeveu) [inline webui/react/src/pages/DesignKit.tsx:249]: Would just call this "SVG Icon" because that's what they are.

**Comment by ClaireNeveu** [comment]
It looks fine in the ui kit but when I go to the new experiment listing the Edit button is still wrong, why is that?

**Comment by johnkim-det** [comment]
> It looks fine in the ui kit but when I go to the new experiment listing the Edit button is still wrong, why is that?

I think the `Dropdown` wrapper is overwriting the classes that are set in a `Button` that's a child element. Looking into it..

**Review by EmilyBonar — APPROVED** [review]
LGTM

**Pull request merged**